go/storage/api/api.go

```diff
@@ -322,6 +324,14 @@ type Backend interface {
 	Initialized() <-chan struct{}
 }
 
+// TODO: extract this out of storage API, into a more general 'PolicyBackend' interface.
```

Comment:
Thanks for the great suggestion and i think i now implemented most of this (so feel free to check again). I still have quite some TODOs in the code so feel free to ignore those, mostly interested in any high level comments. 

A recap of changes since last time:
- added `grpc/proxy` which implements a message agnostic proxy to forward the requests. It's used in `worker/sentry/grpc` so that it forwards all requsts to the upstream node
- added `grpc/auth` which implements server interceptors for authentication. Still need to ckean the existing Polcy implmentation a bit as mentioned in the TODOs

plan to address all the remaining TODOs tomorrow, but that shouldn't be blocking review :)